api4/saml.go

```diff
@@ -247,3 +250,28 @@ func getSamlMetadataFromIdp(c *Context, w http.ResponseWriter, r *http.Request)
 
 	w.Write([]byte(metadata.ToJson()))
 }
+
+func resetAuthDataToEmail(c *Context, w http.ResponseWriter, r *http.Request) {
+	if !c.App.SessionHasPermissionTo(*c.App.Session(), model.PERMISSION_MANAGE_SYSTEM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_SYSTEM)
+		return
+	}
+	type ResetAuthDataParams struct {
+		IncludeDeleted   bool     `json:"include_deleted"`
+		DryRun           bool     `json:"dry_run"`
```

Comment:
Sorry, I missed this before and noticed when reviewing the documentation.  I gave some bad advice. Dry-run should default to "true", but we don't want to default a boolean to true, so we should turn this flag into something like..."execute" with a default it to false. That way they have to set a flag in order to manipulate the database.

Reply:
As discussed offline, we will be keeping the API parameters the same as this would be more intuitive.